### PR TITLE
Vegman bios

### PR DESCRIPTION
**Pull request by nest1ing**

### CODE REVIEW

**Review by artemsen — DISMISSED** [review]

File: src/image_bios.cpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+#include "dbus.hpp"
+#include "fwupderr.hpp"
+#include "subprocess.hpp"
+#include "tracer.hpp"
+
+#include <chrono>
+#include <fstream>
+#include <thread>
+
+static const fs::path aspeedScm = "/sys/bus/platform/drivers/aspeed-smc";
```
Comment (AlexanderAmelkin) [inline src/image_bios.cpp]: `aspeedScm` -> `aspeedSmc`
also, I believe the coding style said that acronyms must be same-case, so it's probably even `aspeedSMC`.

File: src/image_bios.cpp [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+static constexpr auto spiDriver = "1e631000.spi";
+static constexpr auto gpioNamePCHPower = "PWRGD_DSW_PWROK";
+static constexpr auto gpioNameBIOSSel = "MIO_BIOS_SEL";
```
Comment (AlexanderAmelkin) [inline src/image_bios.cpp:22]: Looks like this may be not enough. See the internal PR.

File: src/image_bios.cpp [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        {
+            // Operation completed.
+            return;
+        }
+
+        std::this_thread::sleep_for(
+            std::chrono::milliseconds(delayBetweenChecksMS));
+    }
+
+    throw FwupdateError("The %s SPI driver timed out!",
```
Comment (AlexanderAmelkin) [inline src/image_bios.cpp]: `The SPI driver %s timed out!`

File: src/image_bios.cpp [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        bindSPIDriver();
+
+        locked = true;
+        tracer.done();
+    }
+}
+
+void BIOSUpdater::unlock()
+{
+    if (locked)
```
Comment (AlexanderAmelkin) [inline src/image_bios.cpp:177]: Doesn't look like a lock actually, provided that it doesn't affect any return value or exception throwing.
More like a `bound` flag.
Is flash locking done elsewhere? I mean, what happens if a second copy of `fwupdate` is ran?

**Review by AlexanderAmelkin — CHANGES_REQUESTED** [review]

**Review by alatarum — DISMISSED** [review]
OK, except notes above

File: src/image_bios.cpp (hunk 1 above)
Comment (nest1ing) [reply src/image_bios.cpp]: Done

File: src/image_bios.cpp (hunk 3 above)
Comment (nest1ing) [reply src/image_bios.cpp]: Done

File: src/image_bios.cpp (hunk 2 above)
Comment (nest1ing) [reply src/image_bios.cpp:22]: It seems to me that it is enough. I wrote the detailed answer in the internal PR.

File: src/image_bios.cpp (hunk 4 above)
Comment (nest1ing) [reply src/image_bios.cpp:177]: Yeah, you are right. I missed this aspect.
And two copy of `fwupdate` would disturb each other.
So, I rewrote the code and now use the PCH Power pin as a lock.

**Review by AlexanderAmelkin — APPROVED** [review]

**Pull request merged**